tests/test_qnode.py

```diff
@@ -1398,6 +1348,488 @@ def circuit(a):
             circuit.jacobian([1.0], method="A")
 
 
+class TestMetricTensor:
+    """Tests for metric tensor subcircuit construction and evaluation"""
+
+    def test_no_generator(self):
+        """Test exception is raised if subcircuit contains an
+        operation with no generator"""
+        dev = qml.device('default.qubit', wires=1)
+
+        def circuit(a):
+            qml.Rot(a, 0, 0, wires=0)
```

Comment:
Not important for this PR, but we could break this down into constituent gates and they would all have generators :thinking:

Reply:
Yep definitely. We would need to work out what a PL-level decomposition looks like, since currently it is only handled by the plugins.